lib/core/src/lib/testing/unit-testing-utils.ts

```diff
@@ -0,0 +1,477 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HarnessLoader, TestElement, TestKey } from '@angular/cdk/testing';
+import { DebugElement, Type } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { MatSelectHarness } from '@angular/material/select/testing';
+import { MatOptionHarness } from '@angular/material/core/testing';
+import { MatChipGridHarness, MatChipHarness, MatChipListboxHarness } from '@angular/material/chips/testing';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatIconHarness } from '@angular/material/icon/testing';
+import { MatCheckboxHarness } from '@angular/material/checkbox/testing';
+import { MatErrorHarness, MatFormFieldHarness } from '@angular/material/form-field/testing';
+import { MatInputHarness } from '@angular/material/input/testing';
+import { MatAutocompleteHarness } from '@angular/material/autocomplete/testing';
+import { ComponentFixture } from '@angular/core/testing';
+import { MatTabGroupHarness, MatTabHarness } from '@angular/material/tabs/testing';
+import { MatToolbarHarness } from '@angular/material/toolbar/testing';
+import { MatSnackBarHarness } from '@angular/material/snack-bar/testing';
+import { MatProgressBarHarness } from '@angular/material/progress-bar/testing';
+
+export class UnitTestingUtils {
+    private debugElement: DebugElement;
+    private loader: HarnessLoader;
+
+    constructor(debugElement?: DebugElement, loader?: HarnessLoader) {
+        this.debugElement = debugElement;
+        this.loader = loader;
+    }
+
+    setDebugElement(debugElement: DebugElement): void {
+        this.debugElement = debugElement;
+    }
+
+    setLoader(loader: HarnessLoader): void {
+        this.loader = loader;
+    }
+
+    getByCSS(selector: string): DebugElement {
+        return this.debugElement.query(By.css(selector));
+    }
+
+    getAllByCSS(selector: string): DebugElement[] {
+        return this.debugElement.queryAll(By.css(selector));
+    }
+
+    getInnerTextByCSS(selector: string): string {
+        return this.getByCSS(selector).nativeElement.innerText;
+    }
+
+    getByDataAutomationId(dataAutomationId: string): DebugElement {
+        return this.getByCSS(`[data-automation-id="${dataAutomationId}"]`);
+    }
+
+    getByDataAutomationClass(dataAutomationClass: string): DebugElement {
+        return this.getByCSS(`[data-automation-class="${dataAutomationClass}"]`);
+    }
+
+    getAllByDataAutomationId(dataAutomationId: string): DebugElement[] {
```

Comment:
I'm wondering if there is a sense to have this function because ideally data automation ids should not be repeated. If we have some array then ideally we should have some index/id like: 
data-automation-id="form-input-1"
data-automation-id="form-input-2"
data-automation-id="form-input-3"
....
etc.

Maybe it would have sense to add possibility to change also operator from = to for example ^= (then it search for each element which has data-automation-id starting from for example "form-input")? See doc for others operators: https://www.w3schools.com/css/css_attribute_selectors.asp